src/test/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/actions/manual/ManualTriggerActionApprovalTest.java

```diff
@@ -0,0 +1,225 @@
+
+                ManualTriggerAction.HighLow highLow = action.getCodeReview(json.get(0), 2);
+                assertEquals(2, highLow.getHigh());
+                assertEquals(-1, highLow.getLow());
+
+                highLow = action.getVerified(json.get(0), 2);
+                assertEquals(2, highLow.getHigh());
+                assertEquals(-1, highLow.getLow());
+
+                return null;
+            }
+
+        }).when(session).setAttribute(eq("result"), any());
+
+        action.doGerritSearch("status:open", gerritServerName, false, request, response);
+    }
+
+    /**
+     * Tests {@link ManualTriggerAction#getApprovals(net.sf.json.JSONObject, int)}.
+     * With an all patchsets
+     * @throws Exception if so.
+     */
+    @Test
+    public void testDoGerritSearchAllPatchSets() throws Exception {
+        GerritServer gerritServer = new GerritServer(gerritServerName);
+        PluginImpl.getInstance().addServer(gerritServer);
+        gerritServer.start();
+
+        Config config = (Config)gerritServer.getConfig();
+        config.setGerritAuthKeyFile(sshKey.getPrivateKey());
+        config.setGerritHostName("localhost");
+        config.setGerritSshPort(port);
+        gerritServer.startConnection();
+
+        FreeStyleProject project = DuplicatesUtil.createGerritTriggeredJob(j, projectName, gerritServerName);
+
+        final ManualTriggerAction action = new ManualTriggerAction();
+        HttpSession session = mock(HttpSession.class);
+
+        StaplerRequest request = mock(StaplerRequest.class);
+        StaplerResponse response = mock(StaplerResponse.class);
+
+        when(request.getSession(true)).thenReturn(session);
+
+        doAnswer(new Answer() {
+            /**
+             * @see org.mockito.stubbing.Answer#answer(org.mockito.invocation.InvocationOnMock)
+             */
+            @Override
+            public Object answer(InvocationOnMock aInvocation) throws Throwable {
+                List<JSONObject> json = (List<JSONObject>)aInvocation.getArguments()[1];
+
+                ManualTriggerAction.HighLow highLow = action.getCodeReview(json.get(0), 1);
+                assertEquals(0, highLow.getHigh());
+                assertEquals(0, highLow.getLow());
+
+                highLow = action.getCodeReview(json.get(0), 2);
+                assertEquals(2, highLow.getHigh());
+                assertEquals(-1, highLow.getLow());
+
+                highLow = action.getVerified(json.get(0), 1);
+                assertEquals(0, highLow.getHigh());
+                assertEquals(0, highLow.getLow());
+
+                highLow = action.getVerified(json.get(0), 2);
+                assertEquals(2, highLow.getHigh());
+                assertEquals(-1, highLow.getLow());
+
+                return null;
+            }
+
+        }).when(session).setAttribute(eq("result"), any());
+
+        action.doGerritSearch("status:open", gerritServerName, true, request, response);
```

Comment:
Ah, sorry I probably read it too fast :)

I guess I'm used to seeing some kind of `verify(theMock).someMethod(expected, parameters)` and got confused :)